src/main/java/org/apache/cassandra/distributed/api/SimpleQueryResult.java

```diff
@@ -89,6 +97,12 @@ private SimpleQueryResult(String[] names, Object[][] results, Predicate<Row> fil
         return Collections.unmodifiableList(Arrays.asList(names));
     }
 
+    @Override
+    public List<String> warnings()
+    {
+        return Collections.unmodifiableList(warnings);
```

Comment:
@ifesdjeen I left the defensive wrapping here to protect the internal list, but let me know if this works for you otherwise :)

Reply:
sure, I'd do the same